lsp-lens.el

```diff
@@ -309,6 +309,13 @@ CALLBACK - callback for the lenses."
           (funcall callback lsp-lens--backend-cache lsp--cur-version)
         (lsp-lens--backend-fetch-missing lsp-lens--backend-cache callback lsp--cur-version)))))
 
+(defun lsp--lens-on-refresh (workspace)
+  "Clear lens within all buffers of WORKSPACE, refreshing all workspace buffers."
+  (cl-assert (not (eq nil workspace)))
+  (cl-loop
+   for ws-buffer in (lsp--workspace-buffers workspace) do
+   (lsp-lens-refresh t ws-buffer)))
```

Comment:
can we avoid calling refresh for visible buffers unconditionally, and schedule the refresh once the buffer is visible?

Reply:
I think I missed this comment, sure, that seems a good idea, how can we do that exactly?

Reply:
we have a similar piece of code for flycheck errors. We refresh only the visible buffer. And we should ensure that non visible buffers will refresh when switching to them.